### DISCUSSION TITLE
Displaying previous data while fetching updated data

### DISCUSSION THREAD

**Discussion by ItsMeTazz** [wevm/wagmi#1240] [Q&A]
Hello Wagmi Community, 

Pumped to be around! I'm new to wagmi and already loving it!

I do have a quick question: I might have missed the info somewhere, but is there an integrated way to show the previous data while querying for the most recent one?

Say I'm fetching the hunger from getHunger, and displays 50 in the UI. My UI code would look like this:
` <div>${hunger && hunger.data ? hunger.data : '0.00'}</div>`

Now, I've added the watch keyboard to my useReadContract, which means the data is going to be updated every X seconds. How can I prevent my UI to show '0.00' while the most recent data is being fetched, and keep showing 50 instead? I guess I could work around with another state variable, but just wondering if there's a cleaner way, a wagmi way to do this. 

Appreciate your time and looking forward to reading the answers! 
- Tazz

**Comment by Fabien-Morrow** [comment]
wagmi hooks returns a lot of status variables to monitor the whole fetching process, try to find the ones matching your needs. Maybe with isRefetching ? Never used it, I propose this...
[https://wagmi.sh/docs/hooks/useContractRead](https://wagmi.sh/docs/hooks/useContractRead)

```js
  const hunger= useContractRead({
...myconfig
  })

// (...)

return (
<div>${(hunger.isSuccess || hungrer.isRefetching) ? hunger.data : '0.00' }</div>
)
```
